File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.js
function solution(players, m, k) {
    let answer = 0;
    const server = players.map(_ => 0);
    
    players.forEach((player, time) => {
        const nowServer = (player / m) >> 0;
        const plusServer = nowServer - server[time];
        if (plusServer > 0) {
            answer += plusServer;
            for (let i = time; i < time + k; i++) {
                if (i > 23) break;
                server[i] += plusServer;
            }
        }
    });
    
    return answer;
}

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.py
def solution(players, m, k):
    answer = 0
    server = [0 for _ in range(24)]
    
    for time, player in enumerate(players):
        nowServer = player // m
        plusServer = nowServer - server[time]
        if plusServer > 0:
            answer += plusServer
            for i in range(time, time+k):
                if i > 23: break
                server[i] += plusServer
    
    return answer
